packages/checkout/blocks-registry/register-checkout-block.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+import { registerBlockComponent } from '@woocommerce/blocks-registry';
+import { registerExperimentalBlockType } from '@woocommerce/block-settings';
+
+/**
+ * Internal dependencies
+ */
+import type { CheckoutBlockOptions } from './types';
+import {
+	assertBlockName,
+	assertOption,
+	assertBlockComponent,
+	assertValidArea,
+} from './utils';
+import { registeredBlocks } from './registered-blocks';
+
+/**
+ * Main API for registering a new checkout block within areas.
+ */
+export const registerCheckoutBlock = (
+	blockName: string,
+	options: CheckoutBlockOptions
+): void => {
+	assertBlockName( blockName );
+	assertOption( options, 'areas', 'array' );
+	assertBlockComponent( options, 'component' );
+
+	/**
+	 * This ensures the frontend component for the checkout block is available.
+	 */
+	registerBlockComponent( {
+		blockName,
+		component: options.component,
+	} );
+
+	/**
+	 * If provided with a configuration object, this registers the block with WordPress.
+	 */
+	if ( options?.configuration ) {
+		assertOption( options, 'configuration', 'object' );
+		registerExperimentalBlockType( blockName, {
+			...options.configuration,
+			attributes: {
+				...( options.configuration?.attributes || {} ),
+				...( options.force
+					? {
+							lock: {
+								type: 'object',
+								default: {
+									remove: true,
+								},
+							},
+					  }
+					: {} ),
+			},
+			category: 'woocommerce',
```

Comment:
These blocks are disabled in the large inspector because parent is blank. This groups them in the inner block inspector, which I think is good, but we should let it be overriden.

Reply:
Fixed in https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4671/commits/c27cf0836e42dd73dcfc3292e5f8681630379e9c